sympy/ntheory/elliptic_curve.py

```diff
@@ -60,6 +62,21 @@ def __init__(self, a4, a6, a1=0, a2=0, a3=0, modulus = 0):
         elif isinstance(self._domain, RationalField):
             self._rank = None
 
+        #For Division polynomials
+        X = symbols('X')
+        Y = symbols('Y')
+        div = {}
+        if domain != QQ:
```

Comment:
```suggestion
        if domain is not QQ:
```
`QQ` should be a unique object.